src/MSBuildLocator/MSBuildLocator.cs

```diff
@@ -200,27 +199,17 @@ Assembly TryLoadAssembly(AssemblyName assemblyName)
                 // Assembly resolution is not thread-safe.
                 lock (loadedAssemblies)
                 {
-                    Assembly assembly;
-                    if (loadedAssemblies.TryGetValue(assemblyName.FullName, out assembly))
+                    if (loadedAssemblies.TryGetValue(assemblyName.FullName, out Assembly assembly))
                     {
                         return assembly;
                     }
 
-                    if (IsMSBuildAssembly(assemblyName))
```

Comment:
Delete no-longer-needed `IsMSBuildAssembly` method. It's used in `LoadedMsBuildAssemblies` but I think that should now just return what's currently the local value `loadedAssemblies` (I guess it needs to become instance).